File: csharp/DesiArchitect.DesignPatterns/06_Adapter/Before/OrderShipping.cs
using DesiArchitect.DesignPatterns.Shared;

namespace DesiArchitect.DesignPatterns.Adapter.Before;

// Direct coupling to every partner's unique API shape
public class OrderShipping
{
    public void Ship(string partner, Parcel parcel)
    {
        if (partner == "bluedart")                          // partner-specific branch
        {
            Console.WriteLine($"  [Before] Calling Bluedart.CreateConsignment()");
            Console.WriteLine($"     Ref: {parcel.OrderId}, Dest: {parcel.Address}, Wt: {parcel.WeightKg}kg");
            Console.WriteLine($"     AWB: BD-{parcel.OrderId}-001");
        }
        else if (partner == "delhivery")                    // completely different method shape
        {
            Console.WriteLine($"  [Before] Calling Delhivery.DispatchShipment()");
            Console.WriteLine($"     Waybill: {parcel.OrderId}, Drop: {parcel.Address}, Pkg: {parcel.WeightKg}kg");
            Console.WriteLine($"     Tracking: DL-{parcel.OrderId}-001");
        }
        else if (partner == "ekart")                        // yet another shape
        {
            Console.WriteLine($"  [Before] Calling Ekart.InitiateDelivery()");
            Console.WriteLine($"     Order: {parcel.OrderId}, Addr: {parcel.Address}");
        }
        // New partner = edit this file. One wrong branch = all shipping breaks.
        else
        {
            throw new ArgumentException($"Unknown logistics partner: {partner}");
        }
    }
}

File: typescript/src/06-adapter/before/order-shipping.ts
import type { Parcel } from '../../shared/models.js';

export class OrderShipping {
  ship(partner: string, parcel: Parcel): void {
    if (partner === 'bluedart') {
      console.log('  [Before] Calling Bluedart.CreateConsignment()');
      console.log(`     Ref: ${parcel.orderId}, Dest: ${parcel.address}, Wt: ${parcel.weightKg}kg`);
      console.log(`     AWB: BD-${parcel.orderId}-001`);
    } else if (partner === 'delhivery') {
      console.log('  [Before] Calling Delhivery.DispatchShipment()');
      console.log(`     Waybill: ${parcel.orderId}, Drop: ${parcel.address}, Pkg: ${parcel.weightKg}kg`);
      console.log(`     Tracking: DL-${parcel.orderId}-001`);
    } else if (partner === 'ekart') {
      console.log('  [Before] Calling Ekart.InitiateDelivery()');
      console.log(`     Order: ${parcel.orderId}, Addr: ${parcel.address}`);
    } else {
      throw new Error(`Unknown logistics partner: ${partner}`);
    }
  }
}